pkg/azuredisk/azuredisk_v2.go

```diff
@@ -348,7 +350,7 @@ func (d *DriverV2) StartControllersAndDieOnExit(ctx context.Context) {
 		os.Exit(1)
 	}
 
-	sharedState := controller.NewSharedState(d.Name, d.config.ObjectNamespace, topologyKey, eventRecorder, mgr.GetClient(), d.crdProvisioner.GetDiskClientSet(), d.kubeClient, crdClient)
+	sharedState := controller.NewSharedState(d.Name, d.config.ObjectNamespace, topologyKey, eventRecorder, mgr.GetClient(), d.crdProvisioner.GetDiskClientSet(), d.crdProvisioner.(*provisioner.CrdProvisioner), d.kubeClient, crdClient)
```

Comment:
Can we avoid the cast to the underlying `provisioner.CrdProvisioner` type by adding the `IsDriverUninstall()` method to the `azuredisk.CrdProvisioner` interface? If not, it would be better to remove the `azuredisk.CrdProvisioner` interface and use the `provisioner.CrdProvisioner` type directly in `DriverV2`.

Reply:
Another alternative is to embed the `controller.DriverLifecycle` interface in the `azuredisk.CrdProvisioner` interface.